app/src/main/java/io/homeassistant/companion/android/controls/HaControl.kt

```diff
@@ -49,6 +56,25 @@ interface HaControl {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
             control.setAuthRequired(authRequired)
         }
+        if (entity.attributes["icon"]?.toString()?.startsWith("mdi:") == true &&
+            !entity.attributes["icon"]?.toString()?.substringAfter(":").isNullOrBlank()
+        ) {
+            val iconName = entity.attributes["icon"]!!.toString().split(':')[1]
+            val iconDrawable =
+                IconicsDrawable(context, "cmd-$iconName").apply {
+                    sizeDp = 48
+                }
+            val colorTint = when {
+                entity.domain == "light" && entity.state == "on" -> context.getString(R.color.colorDeviceControlsLightOn)
+                entity.domain == "camera" -> context.getString(R.color.colorDeviceControlsOff)
+                entity.domain == "climate" && entity.state == "heat" -> context.getString(R.color.colorDeviceControlsThermostatHeat)
+                entity.state == "on" -> context.getString(R.color.colorDeviceControlsDefaultOn)
+                entity.state == "off" || entity.state == "unavailable" || entity.state == "unknown" -> context.getString(R.color.colorDeviceControlsOff)
+                else -> context.getString(R.color.colorDeviceControlsDefaultOn)
+            }
+            iconDrawable.setTint(Color.parseColor(colorTint))
```

Comment:
Don't get a color resource as a string 😬 You can simply set `colorTint` to the `R.color.colorDeviceControlsSomething`, and then use `ContextCompat.getColor(context, colorTint)` here instead.